conversion.go

```diff
@@ -613,26 +618,38 @@ func (z *Int) EncodeRLP(w io.Writer) error {
 }
 
 // MarshalText implements encoding.TextMarshaler
+// MarshalText marshals using the decimal representation (compatible with big.Int)
 func (z *Int) MarshalText() ([]byte, error) {
-	return []byte(z.Hex()), nil
+	return []byte(z.Dec()), nil
 }
 
 // MarshalJSON implements json.Marshaler.
+// MarshalJSON marshals using the hexadecimal representation. This is _not_ compatible
+// with big.Int: big.Int marshals into JSON 'native' numeric format.
+//
+// The JSON  native format is, on some platforms, (e.g. javascript), limited to 53-bit large
+// integer space. Thus, U256 uses string-format, which is not compatible with
+// big.int (big.Int refuses to unmarshal a string representation).
+// Since we cannot reach compatibility with big.Int _anyway_, we might aswell
+// choose to use hexadecimal representation here anyway.
```

Comment:
I am unsure about this logic.

I think "the standard" solution employed by most libraries hitting this issue is to return a decimal string. Hex seems very Ethereum specific and I feel this lib should not leak out those decisions at such a low level. Anything against the decimal formatting by default?

Reply:
Changed to decimal